qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/repository/xray/XrayEndpoint.java

```diff
@@ -0,0 +1,51 @@
+package de.qytera.qtaf.xray.repository.xray;
+
+import de.qytera.qtaf.core.QtafFactory;
+import de.qytera.qtaf.core.config.exception.MissingConfigurationValueException;
+import de.qytera.qtaf.xray.config.XrayConfigHelper;
+import de.qytera.qtaf.xray.config.XrayRestPaths;
+
+import java.net.URISyntaxException;
+
+public interface XrayEndpoint {
+
+    /**
+     * Build the 'Authorization' header value for the Xray API.
+     *
+     * @return the authorization header value
+     * @throws MissingConfigurationValueException if no Jira username or API token have been configured
+     */
+    default String getXrayAuthorizationHeaderValue() throws MissingConfigurationValueException, URISyntaxException {
+        if (XrayConfigHelper.isXrayCloudService()) {
+            return getHeaderCloud();
+        }
+        return getHeaderServer();
+    }
+
+    /**
+     * Return the base URL for all Xray endpoints.
+     *
+     * @return the base URL
+     */
+    default String getXrayURL() {
+        if (XrayConfigHelper.isXrayCloudService()) {
+            return XrayRestPaths.XRAY_CLOUD_API_V2;
+        }
+        return XrayConfigHelper.getServerUrl();
+    }
+
```

Comment:
Please add DocString

Reply:
Fixed in [7782f10](https://github.com/Qytera-Gmbh/QTAF/pull/89/commits/7782f10a956c9199bb863187a388462b6c0a5224).